### ISSUE TITLE
Is it possible to add attributes to "instrumentation scope"?

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-java#4473]
[This OTEP](https://github.com/open-telemetry/oteps/pull/201) proposed to add attributes to instrumentation scope.

This issue is a question for the maintainers of this repo: are we able to do this in a non-breaking manner and without significant performance impact?

**Comment by jack-berg** [comment]
Yes, I believe so. @anuraaga / @jkwatson I imagine scope attributes would be settable on `MeterBuilder` / `TracerBuilder` / `LogEmitterBuilder` via a new method `setAttributes(Attributes)`. Attributes will default to empty. When [ComponentRegistry#get](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/internal/ComponentRegistry.java#L69-L86) is called to obtain a meter / tracer / logemitter, it will have to evaluate `InstrumentationScopeInfo` equality with attributes, in addition to the scope name, version, and schema URL. 

In most cases attributes will be empty and the addition of attributes to `InstrumentationScopeInfo` will have minimal performance impact. In cases where attributes are not empty, performance will dictated by the performance of `Attributes.equals()` which I understand is reasonably good.

**Comment by jkwatson** [comment]
Yep. The builders are the place to add it to the API, for sure.  In the SDK, we'll need to add some sort of creational method for `InstrumentationScopeInfo`, but I'm sure we can cook something up for that.

**Comment by tigrannajaryan** [comment]
Thanks, @jack-berg @jkwatson
